generator/internal/metadata_decorator_generator.cc

```diff
@@ -126,19 +126,20 @@ Status MetadataDecoratorGenerator::GenerateHeader() {
   HeaderLocalIncludes({vars("stub_header_path"), "google/cloud/version.h"});
   HeaderSystemIncludes(
       {HasLongrunningMethod() ? "google/longrunning/operations.grpc.pb.h" : "",
-       "memory", "string"});
+       "memory", "string", "unordered_map"});
 
   auto result = HeaderOpenNamespaces(NamespaceType::kInternal);
   if (!result.ok()) return result;
 
   // metadata decorator class
-  HeaderPrint(  // clang-format off
-    "\n"
-    "class $metadata_class_name$ : public $stub_class_name$ {\n"
-    " public:\n"
-    "  ~$metadata_class_name$() override = default;\n"
-    "  explicit $metadata_class_name$(std::shared_ptr<$stub_class_name$> child);\n");
-  // clang-format on
+  HeaderPrint(R"""(
+class $metadata_class_name$ : public $stub_class_name$ {
+ public:
+  ~$metadata_class_name$() override = default;
+  explicit $metadata_class_name$(
+      std::shared_ptr<$stub_class_name$> child,
+      std::unordered_map<std::string, std::string> fixed_metadata = {});
```

Comment:
The `grpc::ClientContext` map is actually a `std::multimap<std::string, std::string>`.  We should probably match that.

Reply:
Good catch. Fixed.